lib/ui/page/home/page/chat/widget/message_info/controller.dart

```diff
@@ -17,61 +17,155 @@
 import '/domain/repository/user.dart';
-import '/domain/service/user.dart';
-import '/ui/widget/text_field.dart';
+import '/domain/service/chat.dart';
+import '/store/chat_rx.dart';
+import '/store/model/chat_item.dart';
+import '/util/message_popup.dart';
+import '/util/obs/obs.dart';
 
 /// Controller of the [MessageInfo] popup.
 class MessageInfoController extends GetxController {
-  MessageInfoController(this._userService, {this.reads = const []});
+  MessageInfoController(this.chatId, this.chatItemId, this._chatService);
 
-  /// [LastChatRead]s who read the [ChatItem] this [MessageInfo] is about.
-  final Iterable<LastChatRead> reads;
+  /// ID of the [Chat] this page is about.
+  final ChatId? chatId;
+
+  /// ID of the [ChatItem] this page is about.
+  final ChatItemId? chatItemId;
+
+  /// [DtoChatItem] of a [RxChat] this [MessageInfo] is about.
+  final Rx<DtoChatItem?> chatItem = Rx(null);
+
+  /// [RxUser]s who may read [ChatItem] this [MessageInfo] is about.
+  final RxList<RxUser> members = RxList();
+
+  /// [LastChatRead]s of a [ChatItem] this [MessageInfo] is about.
+  RxList<LastChatRead> reads = RxList();
+
+  /// Specifies whether to display the status of all [RxChat.members].
+  final Rx<bool?> displayMembers = Rx(null);
 
   /// [ScrollController] to pass to a [Scrollbar].
   final ScrollController scrollController = ScrollController();
 
-  /// [TextFieldState] of the search field.
-  final TextFieldState search = TextFieldState();
+  /// Indicates whether the [RxChat.members] have a next page.
+  RxBool get haveNext => _chat?.members.hasNext ?? RxBool(false);
+
+  /// Reactive state of the [RxChat] this page is about.
+  RxChat? _chat;
 
-  /// Reactive value of the [search] field.
-  final RxString query = RxString('');
+  /// [Chat]s service used to get the [_chat] value.
+  final ChatService _chatService;
 
-  /// [RxUser]s of the [reads].
-  final RxList<RxUser> users = RxList();
+  /// Indicator whether the [_scrollListener] is already invoked during the
+  /// current frame.
+  bool _scrollIsInvoked = false;
 
-  /// [UserService] fetching the [users].
-  final UserService _userService;
+  /// Subscription for the [RxChat.members] changes.
+  StreamSubscription? _membersSubscription;
 
   @override
   void onInit() {
-    _fetchUsers();
     super.onInit();
+
+    scrollController.addListener(_scrollListener);
+    _initChat();
   }
 
   @override
   void onClose() {
+    _membersSubscription?.cancel();
     scrollController.dispose();
+
     super.onClose();
   }
 
-  /// Fetches the [users] from the [UserService].
-  Future<void> _fetchUsers() async {
-    final List<Future<void>> futures = reads
-        .map((read) async {
-          final FutureOr<RxUser?> fetched = _userService.get(read.memberId);
-          final RxUser? user = fetched is RxUser? ? fetched : await fetched;
-          if (user != null) {
-            users.add(user);
+  /// Init [RxChat] and [DtoChatItem].
+  Future<void> _initChat() async {
```

Comment:
This documentation doesn't follow the Effective Dart: Documentation recommendations:

https://dart.dev/effective-dart/documentation#prefer-starting-comments-of-a-function-or-method-with-third-person-verbs-if-its-main-purpose-is-a-side-effect